addons/survey/static/src/js/survey.js

```diff
@@ -161,8 +165,38 @@ if(!the_form.length) {
     //         console.debug("[survey] Focus lost on question " + $(this).attr("id"));
     // });
 
-    // Launch prefilling
-    prefill();
+    function load_locale(){
+        var url = "/web/webclient/locale/" + base.get_context().lang || 'en_US';
+        return ajax.loadJS(url);
+    }
+
+    var ready_with_locale = $.when(base.ready(), load_locale());
+    // datetimepicker use moment locale to display date format according to language
+    // frontend does not load moment locale at all.
+    // so wait until DOM ready with locale then init datetimepicker
+    ready_with_locale.then(function(){
+        var l10n = _t.database.parameters;
+        $('.form-control.date').datetimepicker({
+            format : time.strftime_to_moment_format(l10n.date_format),
+            minDate: moment({ y: 1900 }),
+            maxDate: moment().add(200, "y"),
+            calendarWeeks: true,
+            icons: {
+                time: 'fa fa-clock-o',
+                date: 'fa fa-calendar',
+                next: 'fa fa-chevron-right',
+                previous: 'fa fa-chevron-left',
+                up: 'fa fa-chevron-up',
+                down: 'fa fa-chevron-down',
+            },
+            locale : moment.locale(),
+            allowInputToggle: true,
+            keyBinds: null,
+        });
+
+        // Launch prefilling
+        prefill();
```

Comment:
I am not totally sure, but maybe the prefill() call should be out of the deferred.  It does not need the locale to do its job, and putting it here means that it will take slightly longer for the form to be prefilled.